django/university/exchange/utils.py

```diff
@@ -0,0 +1,266 @@
+            return (ExchangeStatus.STUDENTS_NOT_ENROLLED, None)
+
+        # print("Class from sigarra: ", get_class_from_sigarra(schedule[auth_student][(class_auth_student_goes_from, course_unit)]["ocorrencia_id"], class_auth_student_goes_to, cookies)[0])
+
+        schedule[auth_student][(class_auth_student_goes_to, course_unit)] = SigarraController().get_class_schedule(schedule[auth_student][(class_auth_student_goes_from, course_unit)]["ocorrencia_id"], class_auth_student_goes_to).data[0][0]# get class schedule
+        del schedule[auth_student][(class_auth_student_goes_from, course_unit)] # remove old class of other student
+
+    return (ExchangeStatus.SUCCESS, None) 
+
+def get_unit_schedule_url(ocorrencia_id, semana_ini, semana_fim):
+    return f"https://sigarra.up.pt/feup/pt/mob_hor_geral.ucurr?pv_ocorrencia_id={ocorrencia_id}&pv_semana_ini={semana_ini}&pv_semana_fim={semana_fim}"
+
+def build_new_schedules(student_schedules, exchanges, auth_username):
+    for curr_exchange in exchanges:
+        print("Other student is: ", curr_exchange["other_student"])
+        print("Auth student is: ", auth_username)
+        other_student = curr_exchange["other_student"]["mecNumber"]
+        course_unit = CourseUnit.objects.get(pk=curr_exchange["courseUnitId"])
+        course_unit = course_unit.acronym
+        class_auth_student_goes_to = curr_exchange["classNameRequesterGoesTo"]
+        class_other_student_goes_to = curr_exchange["classNameRequesterGoesFrom"] # The other student goes to its new class
+
+        print("auth student goes to: ", class_auth_student_goes_to)
+        print("other student goes to: ", class_other_student_goes_to)
+        
+        print("what in the hell? ", student_schedules[other_student])
+        print("course unit: ", course_unit)
+        
+        # If participant is neither enrolled in that course unit or in that class
+        other_student_valid = (class_auth_student_goes_to, course_unit) in student_schedules[other_student]
+        auth_user_valid = (class_other_student_goes_to, course_unit) in student_schedules[auth_username]
+        
+        print("other studenet valid: ", other_student_valid)
+        print("auth studenet valid: ", auth_user_valid)
+
+        if not(other_student_valid) or not(auth_user_valid):
+            return (ExchangeStatus.STUDENTS_NOT_ENROLLED, None)
+
+        # Change schedule
+        tmp = student_schedules[auth_username][(class_other_student_goes_to, course_unit)]
+        student_schedules[auth_username][(class_auth_student_goes_to, course_unit)] = student_schedules[other_student][(class_auth_student_goes_to, course_unit)]
+        student_schedules[other_student][(class_other_student_goes_to, course_unit)] = tmp
+
+        del student_schedules[other_student][(class_auth_student_goes_to, course_unit)] # remove old class of other student
+        del student_schedules[auth_username][(class_other_student_goes_to, course_unit)] # remove old class of auth student
```

Comment:
This is not automatically understanding code. Can you try to refactor a little?

Reply:
@thePeras I refactored that code to this:

```python
# Assign the class the auth student is going to to its schedule
schedule_of_auth_student_new_class = student_schedules[other_student][(class_auth_student_goes_to, course_unit)]
student_schedules[auth_username][(class_auth_student_goes_to, course_unit)] = schedule_of_auth_student_new_class

# Assign the class the other student is going to to its schedule
schedule_of_other_student_new_class = student_schedules[auth_username][(class_other_student_goes_to, course_unit)]
student_schedules[other_student][(class_other_student_goes_to, course_unit)] = schedule_of_other_student_new_class

# Remove class the other student is going from and will not be in anymore
del student_schedules[other_student][(class_auth_student_goes_to, course_unit)]

# Remove class the auth student is going from and will not be in anymore
del student_schedules[auth_username][(class_other_student_goes_to, course_unit)]
```

Do you think it is clear enough now?

Reply:
Not really, check my new suggestion using tuple swap!

Reply:
I've now changed the code to switch to tuple swap